sdk/src/main/java/com/google/cloud/dataflow/sdk/options/GcpOptions.java

```diff
@@ -183,24 +184,67 @@ void setCredentialFactoryClass(
   public static class DefaultProjectFactory implements DefaultValueFactory<String> {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultProjectFactory.class);
 
+    private static String getUserHome() {
+      return System.getProperty("user.home");
+    }
+
+    private String getActiveConfig() throws IOException {
+      final String defaultConfigName = "default";
+
+      // get active configuration
+      File activeConfFile = new File(
+          getUserHome(),
+          ".config/gcloud/active_config");
```

Comment:
It seems as though Google is vending a library that already does this and instead of Dataflow owning this logic, we can rely on this library but wanted to make sure that this meets your needs:

https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/cloud/ServiceOptions.java
Look at the logic for defaultProject(), specifically activeGoogleCloudConfig() and googleCloudProjectId()